controller/proxy-injector/server.go

```diff
@@ -76,24 +77,23 @@ func (w *WebhookServer) Shutdown() error {
 	return w.Server.Shutdown(context.Background())
 }
 
-func tlsConfig(certFile, keyFile string) (*tls.Config, error) {
-	certBytes, err := ioutil.ReadFile(certFile)
-	if err != nil {
-		return nil, err
-	}
-
-	keyBytes, err := ioutil.ReadFile(keyFile)
-	if err != nil {
-		return nil, err
+func tlsConfig(rootCA *pkgTls.CA, controllerNamespace string) (*tls.Config, error) {
+	tlsIdentity := k8s.TLSIdentity{
+		Name:                "linkerd-proxy-injector",
+		Kind:                k8s.Service,
+		Namespace:           controllerNamespace,
+		ControllerNamespace: controllerNamespace,
 	}
+	dnsName := tlsIdentity.ToDNSName()
```

Comment:
What DNS name does this evaluate to?  `linkerd-proxy-injector.<controllerNS>.svc`?  Is that the DNS name that the k8s API will call this by?

Reply:
That's right. Here's the k8s piece that prepares the client for that call:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/webhook/client.go#L151